networkx/convert_matrix.py

```diff
@@ -432,7 +432,11 @@ def from_pandas_edgelist(
     g = nx.empty_graph(0, create_using)
 
     if edge_attr is None:
-        g.add_edges_from(zip(df[source], df[target]))
+        if g.is_multigraph() and edge_key is not None:
+            for u, v, k in zip(df[source], df[target], df[edge_key]):
+                g.add_edge(u, v, k)
```

Comment:
Could this use `add_edges_from` like the else clause?
```python
            g.add_edes_from(zip(df[source], df[target], df[edge_key])
```

Reply:
Reasonable suggestion. I tried that way. I think using `add_edges_from` here would be significantly slower:
```python
In [1]: import pandas as pd

In [2]: import networkx as nx

In [3]: L0 = list(range(10000))

In [4]: L1 = list(range(1, 10001))

In [5]: K = list(range(10000, 20000))

In [6]: df = pd.DataFrame({'a': L0, 'b': L1, 'c': K})

In [7]: %%timeit
   ...: G = nx.MultiGraph()
   ...: for u, v, k in zip(df['a'], df['b'], df['c']):
   ...:     G.add_edge(u, v, k)
   ...:
13.1 ms ± 19 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [8]: %%timeit
   ...: G = nx.MultiGraph()
   ...: G.add_edges_from(zip(df['a'], df['b'], df['c']))
   ...:
   ...:
25.9 ms ± 136 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [9]: %%timeit
   ...: G = nx.MultiDiGraph()
   ...: for u, v, k in zip(df['a'], df['b'], df['c']):
   ...:     G.add_edge(u, v, k)
   ...:

14.4 ms ± 173 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [10]: %%timeit
    ...: G = nx.MultiDiGraph()
    ...: G.add_edges_from(zip(df['a'], df['b'], df['c']))
    ...:
    ...:
27.9 ms ± 164 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
Ahh... Thanks for that -- I knew there was technical debt in the multi(di)graph `add_edges_from`. If I recall correctly, there is a tradeoff between rewriting `add_edges_from` for each base class and speed up updates. And there was some discounting of time issues for multiedges since they were a niche feature at the time. Let's get those up to speed!

But that's a different PR.  This one is fine as is.   Thanks!!